Examples/Framework/include/ActsExamples/EventData/GeometryContainers.hpp

```diff
@@ -69,6 +83,21 @@ struct GeometryIdGetter {
 struct CompareGeometryId {
   // indicate that comparisons between keys and full objects are allowed.
   using is_transparent = void;
+
+  /// @brief Comparator that sorts objects with the same geometry id by time
+  template <TimedObj Left, TimedObj Right>
```

Comment:
So this is essentially a staged comparison, where the time is only used to enforce an ordering if the geometry ids are already found to be the same, yes?

Maybe this should be a separate comparator type? OTOH, this is probably not very important.